src/main/java/Duke.java

```diff
@@ -1,10 +1,357 @@
+                }
+                break;
+            case COMMAND_UNMARK:
+                try {
+                    if (taskDesc.length() == 0) {
+                        throw new EmptyDescriptionException();
+                    }
+                    taskNum = Integer.parseInt(inputText[1]);
+                    if (taskNum > list.size() || taskNum <= 0) {
+                        throw new ArrayIndexOutOfBoundsException();
+                    }
+                    unmarkTask(taskNum);
+                } catch (EmptyDescriptionException e) {
+                    e.emptyDescriptionNumber();
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println(DIVIDER + System.lineSeparator()
+                            + "Unable to unmark task as task does not exist!" + System.lineSeparator() + DIVIDER);
+                } catch (NumberFormatException e) {
+                    System.out.println(DIVIDER + System.lineSeparator()
+                            + "You did not key in a task number. Please key in a valid task number to unmark!"
+                            + System.lineSeparator() + DIVIDER);
+                }
+                break;
+            case COMMAND_TODO:
+                try {
+                    if (taskDesc.length() == 0) {
+                        throw new EmptyDescriptionException();
+                    }
+                    list.add(new Todo(taskDesc));
+                    acknowledgementMessage();
+                } catch (EmptyDescriptionException e) {
+                    e.emptyDescriptionTodo();
+                }
+                break;
+            case COMMAND_EVENT:
+                try {
+                    if (taskDesc.length() == 0) {
+                        throw new EmptyDescriptionException();
+                    }
+                    //use string.split to split the string into their different descriptions
+                    String[] eventInput = taskDesc.split(COMMAND_FROM);
+                    //split into task description and duration
+                    String eventTaskDesc = eventInput[0];
+                    String eventDuration = eventInput[1];
+                    String[] eventStartAndEnd = eventDuration.split(COMMAND_TO);
+                    String eventStart = eventStartAndEnd[0];
+                    String eventEnd = eventStartAndEnd[1];
+                    list.add(new Event(eventTaskDesc, eventStart, eventEnd));
+                    acknowledgementMessage();
+                } catch (EmptyDescriptionException e) {
+                    e.emptyDescriptionEvent();
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    System.out.println(DIVIDER + System.lineSeparator()
+                            + "The event you keyed in was invalid!" + System.lineSeparator()
+                            + "Please key in a valid event!" + System.lineSeparator() + DIVIDER);
```

Comment:
Print statements like this could be handled seperately in a message handler class instead of a println statement here